<launch>
  <arg name="viz" default="false"/>
  <arg name="service_name" default="/run_inference"/>
  <arg name="scan_topic" default="/point_scan"/>
  <arg name="controller_freq" default="$(optenv GM_PARAM_CONTROLLER_FREQ 5)"/>
  <arg name="global_planning_freq" default="$(optenv GM_PARAM_GLOBAL_PLANNING_FREQ 1)"/>
  
  <node name="egocircle" pkg="egocircle" type="ego_circle"/>

  <include file="$(find turtlebot_navigation)/launch/includes/velocity_smoother.launch.xml"/>
  
  <node name="move_base" pkg="move_base" type="move_base" output="screen">
    <rosparam file="$(find nav_configs)/config/common_costmap_params.yaml" command="load" ns="global_costmap"/>
    <rosparam file="$(find nav_configs)/config/common_costmap_params.yaml" command="load" ns="local_costmap"/>
    <rosparam file="$(find nav_configs)/config/local_costmap_params.yaml" command="load"/>
    <rosparam file="$(find nav_configs)/config/global_costmap_params.yaml" command="load"/>

    <rosparam file="$(find nav_configs)/config/local_planner_common_params.yaml" command="load" ns="DWAPlannerROS"/>
    <rosparam file="$(find nav_configs)/config/move_base_params.yaml" command="load"/>
    <rosparam file="$(find nav_configs)/config/global_planner_params.yaml" command="load" ns="GlobalPlanner"/>

    <param name="controller_frequency" value="$(arg controller_freq)"/>
    <param name="planner_frequency" value="$(arg global_planning_freq)"/>

    <!--<remap from="/cmd_vel" to="/cmd_vel_mux/input/navi"/>-->
    <remap from="cmd_vel" to="navigation_velocity_smoother/raw_cmd_vel"/>

    <!-- Approach-specific parameters/remappings/etc -->
    <param name="base_local_planner" value="p2d_local_planner/DWAPlannerROS"/>
    <rosparam file="$(find nav_configs)/config/pips_dwa_local_planner_common_params.yaml" command="load" ns="DWAPlannerROS"/>

    <param name="DWAPlannerROS/scan_topic" value="$(arg scan_topic)"/>
    <param name="DWAPlannerROS/service_name" value="$(arg service_name)"/>
  </node>

  <arg name="model_path" default="$(find dl_models)/downloads/laser-local-goal-cmd-local_general-0.pb"/>
  <include file="$(find p2d_ros)/launch/inference_service.launch">
    <arg name="model_path" value="$(arg model_path)"/>
  </include>

  <group if="$(arg viz)">
    <node name="goal_viz" pkg="p2d_ros" type="goal_pub.py"/>
    
    <node name="twist_viz" pkg="pips_visualization" type="twist_viz.py" >
      <remap from="twist" to="/cmd_vel_mux/input/navi"/>
      <remap from="path_out" to="viz/twist"/>
      <param name="sim_time" value="5"/>
    </node>
  
  </group>
  

  

</launch>
